File: Getting Started Storyboard Objective-C/Getting Started Storyboard Objc/ViewController.m
//
//  ViewController.m
//  Getting Started Storyboard Objc
//
//  Created by Guy Daher on 23/06/2017.
//  Copyright © 2017 Guy Daher. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    [InstantSearch.shared registerAllWidgetsIn:self.view];
    
    self.hitsController = [[HitsController alloc] initWithTable: self.tableView];
    self.tableView.dataSource = self.hitsController;
    self.tableView.delegate  = self.hitsController;
    self.hitsController.tableDataSource = self;
}

-(UITableViewCell*) tableView:(UITableView *)tableView cellForRowAt:(NSIndexPath *)indexPath containing:(NSDictionary<NSString *,id> *)hit {
    
    UITableViewCell* cell = [tableView dequeueReusableCellWithIdentifier:@"hitCell" forIndexPath:indexPath];
    
    cell.textLabel.highlightedTextColor = [UIColor blueColor];
    cell.textLabel.highlightedBackgroundColor = [UIColor yellowColor];
    
    
    cell.textLabel.highlightedText = [[SearchResults highlightResultWithHit:hit path:@"name"] value];
    
    return cell;
}


@end

File: Getting Started Storyboard/Getting Started Storyboard/ViewController.swift
//
//  ViewController.swift
//  Getting Started Storyboard
//
//  Created by Guy Daher on 01/06/2017.
//  Copyright © 2017 Guy Daher. All rights reserved.
//

import UIKit
import InstantSearch

class ViewController: HitsTableViewController {

    @IBOutlet weak var tableView: HitsTableWidget!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        hitsTableView = tableView
        
        InstantSearch.reference.addAllWidgets(in: self.view)
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath, containing hit: [String : Any]) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "hitCell", for: indexPath)
        
        cell.textLabel?.text = hit["name"] as? String
        cell.detailTextLabel?.text = String(hit["salePrice"] as! Double)
        
        return cell
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

